docs/architecture/ADR-006-oracle.md

```diff
@@ -0,0 +1,69 @@
+# ADR 006: Oracle
+
+## Changelog
+
+- November 29th, 2021: Initial Draft (@adamewozniak)
+
+## Status
+
+Proposed
+
+## Context
+
+Umee needs an oracle to determine the price of assets.
+
+From section 5.1 of the [Umee Whitepaper](https://umee.cc/umee-whitepaper/):
+
+> Oracle reads asset price and updates the state to the Asset Facility Coordination
+Protocol
+
+We've already decided to import a large chunk of this from [Terra's oracle](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html), although a few parts of this are specific to Terra's protocol and do not need to be implemented.
+
+## Alternative Approaches
+
+- Cloning the x/oracle module completely. This would leave our code dirty, and we'd later have issues interfacing with Terra's [Cross Exchange Rate](https://docs.terra.money/Reference/Terra-core/Module-specifications/spec-oracle.html#compute-cross-exchange-rate-using-reference-terra), since it's designed for getting the exchange rate of only Terra.
+- Using something like Band or Chainlink. This would be additional overhead, and we'd [have less control](https://github.com/umee-network/umee/issues/97#issuecomment-923914840) over how our oracle works.
```

Comment:
I still don't know or understand how we'd use an external Oracle service like Band or Chainlink. Do you know how?

Reply:
I'm not sure about band, but I believe Chainlink is specifically built for smart contracts, so I don't think we could even use it for validator modules - [chainlink api](https://docs.chain.link/docs/request-and-receive-data/)

Reply:
Looks like [somebody did it](https://betterprogramming.pub/connect-a-chainlink-oracle-to-a-cosmos-blockchain-d7934d75bae5) using chainlink's [external initiators](https://docs.chain.link/docs/external-initiators-introduction/), although this seems more convoluted than building our own for what we specifically need

Reply:
Added a couple references to that implementation in the adr